zyte_common_items/pipelines.py

```diff
@@ -53,3 +57,43 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropItem(Exception):
+    """Drop item from the item pipeline."""
+
+    pass
+
+
+class DropLowProbabilityItemPipeline:
+    """Drop item with a probability less than threshold."""
+
+    def __init__(self, stats):
+        self.stats = stats
+        self.threshold = None
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        o = cls(crawler.stats)
+        return o
+
+    def get_threshold(self, item, spider):
+        return spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {}).get(
+            global_object_name(item.__class__), DEFAULT_ITEM_PROBABILITY_THRESHOLD
+        )
+
+    async def process_item(self, item, spider):
+        if not self.threshold:
+            self.threshold = self.get_threshold(item, spider)
```

Comment:
Doesn’t this mean that only the first item class threshold will be taken into account, if a spider returns 2+ types of items?

Reply:
Oh, yes, this was the initial idea. Is it possible that the spider can produce multiple types of items?

Reply:
> Is it possible that the spider can produce multiple types of items?

If we consider custom, user-made subclasses, definitely. But also I wonder if, in the future, we might support e.g. crawling product reviews in addition to product themselves. I think it might be better to assume it will happen eventually and be ready for it, if it does not complicate things too much.

Reply:
Ahh, got it, makes sense. Thanks!